spec/models/weight_limits_spec.rb

```diff
@@ -91,6 +101,171 @@ def build_inventory_unit(variant, order)
       Spree::ActiveShipping::Config.set(:default_weight => 1)
     end
 
+    describe ".valid_weight_for_package?" do
+      context "with a non nil max weight" do
+        it "should return true if the max_weight is equal to zero" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 0).to eq true
+        end
+
+        it "should return true if the package weight is equal to the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 1).to eq true
+        end
+
+        it "should return true if the package weight is less than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq true
+        end
+
+        it "should return false if the package weight is greater than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(3)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq false
+        end
+      end
+
+      context "with a nil max weight" do
+        it "should return false if the max_weight is nil" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, nil).to eq false
+        end
+      end
+    end
+
+    describe ".country_weight_error?" do
```

Comment:
Same as above, this should be tested as part of the tests for `ActiveShipping::Base#available?`.